cmd/ccboc/cmd/commands.go

```diff
@@ -0,0 +1,278 @@
+		Token:  token,
+	}
+
+	if err := cfg.Validate(); err != nil {
+		return fmt.Errorf("config validation failed: %v", err)
+	}
+
+	user, err := user.Current()
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	home := user.HomeDir
+
+	data, err := json.MarshalIndent(cfg, "", "")
+	if err != nil {
+		return fmt.Errorf("couldn't marthal the configuration file: %v", err)
+	}
+
+	path := filepath.Join(home, config.DefaultConfigPath)
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		if err := os.Mkdir(path, os.ModePerm); err != nil {
+			return fmt.Errorf("couldn't create dir: %v", err)
+		}
+	}
+
+	configFilePath := filepath.Join(home, config.DefaultConfigPath, config.DefaultConfigFileName)
+	if err := ioutil.WriteFile(configFilePath, data, 0777); err != nil {
+		return fmt.Errorf("couldn't write to file: %v", err)
+	}
+
+	fmt.Printf("Configuration file generated at %s\n", configFilePath)
+	return nil
+}
+
+func getCalculations() error {
+	body, responseError, err := request("GET", globalConfig.APIURL+"/calculations", bytes.NewBuffer(nil))
+	if err != nil {
+		logrus.WithError(err).Fatal("error while perfoming the http request")
+	} else if responseError != nil {
+		logrus.WithFields(logrus.Fields{"message": responseError.Message, "status_code": responseError.StatusCode}).Fatal("errors occurred")
+	}
+
+	var data *calculationsv1.CalculationList
+	if err := json.Unmarshal(body, &data); err != nil {
+		logrus.WithError(err).Fatal("Error unmarshaling json response")
+	}
+	output(data)
+	return nil
+}
+
+func getCalculationID() error {
+	args := os.Args
+	calcID := args[len(args)-1]
+
+	body, responseError, err := request("GET", globalConfig.APIURL+"/calculation/"+calcID, bytes.NewBuffer(nil))
+	if err != nil {
+		logrus.WithError(err).Fatal("error while perfoming the http request")
+	} else if responseError != nil {
+		logrus.WithFields(logrus.Fields{"message": responseError.Message, "status_code": responseError.StatusCode}).Fatal("errors occurred")
+	}
+
+	var calc *calculationsv1.Calculation
+	if err := json.Unmarshal(body, &calc); err != nil {
+		logrus.WithError(err).Fatal("Error unmarshaling json response")
+	}
+
+	output(calc)
+	return nil
+}
+
+func getCalculationResult() error {
+	req, err := http.NewRequest(http.MethodGet, globalConfig.APIURL+"/calculations/results?teff="+fmt.Sprintf("%0.1f", teff)+"&logg="+fmt.Sprintf("%0.2f", logG), bytes.NewBuffer(nil))
+	if err != nil {
+		logrus.WithError(err).Fatal("Coudln't create the request.")
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", globalConfig.Token))
+	req.Header.Add("Accept", "application/json")
+
+	client := &http.Client{Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}}
+	resp, err := client.Do(req)
+	if err != nil {
+		logrus.WithError(err).Fatal("Coudln't fullfil the request.")
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error while reading the response.")
```

Comment:
All those `Fatal`s should be removed and instead return the error.

Reply:
There are 22 of these, all of them?

Reply:
Yea. the functions should return the error, and we fatal when a commands returned a non-nil error

Reply:
How could we then differentiate all the errors? Wouldn't that just make the user experience much worse, when they don't know what actually happened?

Reply:
We are not changing anything. It's better practice for the func to return an error rather than allowing logrus to os.Exit the app.
Once we return the error we can output it and exit.